monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -276,7 +276,10 @@ def get_all_case_stats(self, key="training", transform_list=None):
         result[DataStatsKeys.BY_CASE] = [None] * n_cases
```

Comment:
perhaps we keep the yaml filename instead of null to avoid confusion...
```suggestion
        result[DataStatsKeys.BY_CASE] = self.output_path.replace(".yaml", "_by_case.yaml")
```

Reply:
```
[None] * n_cases
```
looks like a workaround to pass this line https://github.com/Project-MONAI/MONAI/blob/eed660c0a949290bb9aa2541f6751575bafeab72/tests/test_auto3dseg.py#L191

Reply:
Maybe we should just change the test instead of adding the key with `none` values?

Reply:
sorry, ignore my previous comment, this is testing the `result` variable's `"stats_by_cases"` key, instead of the content of the yaml file.  

the variable is still needed by FL https://github.com/Project-MONAI/MONAI/blob/eed660c0a949290bb9aa2541f6751575bafeab72/monai/fl/client/monai_algo.py#L275

Reply:
Please keep the list of None, it's for compatibility, our old algos look at len(case_by_stat) to estimate the length of dataset